src/nrtk_explorer/app/transforms.py

```diff
@@ -139,15 +135,13 @@ def on_apply_transform(self, *args, **kwargs):
         transform = self._transforms[current_transform]
         for image_id in self.state.source_image_ids:
             image = self.context["image_objects"][image_id]
-
             transformed_image_id = f"transformed_{image_id}"
-
             transformed_img = transform.execute(image)
-
+            if image.size != transformed_img.size:
+                # Resize so pixel-wise annotation similarity score works
+                transformed_img = transformed_img.resize(image.size)
```

Comment:
What impact does a downsample with whatever filtering method PIL picks have on the object detection model output.  Can we ask the perturber to match our input image size?

Reply:
IT depends on the object dect model, it can be significant since due the change of the embeddings. 

One thing to note is that we need to group the images by image size for batching to work.

> Can we ask the perturber to match our input image size?

In principle we want to check the perturbation in the model meaning that we should not resize so we can see how the downsampling affects.

Reply:
In the case where the transformed image is a different resolution should we resize to original or keep the transformed image size?  We could resize just the annotations before passing to the `scorer`.

(Aside: I see we are resizing all images passed to the embedding plot, not the object detection model.  https://github.com/Kitware/nrtk-explorer/blob/main/src/nrtk_explorer/app/embeddings.py#L130 )